### PR TITLE
Add update-shell command

### LINKED ISSUE #8: Implement a way to update development shells
The UX could be:

* `toolbox update-shell`
* `toolbox make-shell --update`

Implementation:

* rerun `toolbox make-shell` with the dependencies specified the last time:
  * evaluate `shell.nix` and pull out `buildInputs`
  * parse the first line `Generated by: ...`
  * cons: we loose possible user customization
* modify only the toolbox commit/sha256 with a regexp or something like that
  * cons: might be hard if other sources are pulled in
* on make-shell generate a JSON file with toolbox commit+sha256 that is used in `shell.nix`
  * on update just the JSON must be updated
  * easy because the JSON is structured

### PR DESCRIPTION
**Pull request by eonpatapon**
* `make-shell` now creates at `toolbox.json` file that references the
  `commit` and `sha256` used for the current shell
* `update-shell` by default will update the `toolbox.json` to the latest
  master commit
* `update-shell` can take a specific commit as argument

Closes #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,18 +114,24 @@ You can do it with two methods:
...
-Run again in you project:
+To update the shell to the latest toolbox version, run:
 
 ```sh
-toolbox make-shell terraform ansible
+toolbox update-shell
 ```
 
-This will update `shell.nix` with the most recent commit of the toolbox.
+This will update `toolbox.json` with the most recent commit of the toolbox.
```
Comment (jeci-sarl) [inline README.md]: last commit on the master branch

File: scripts/commands.sh [hunk 2]
```diff
@@ -76,64 +75,72 @@ doctor() {
...
-        instantiate_args="$instantiate_args -A $tool"
-    done
-    result=$(nix-instantiate $url $instantiate_args 2>&1 || true)
-    if echo $result | grep -q error; then
-        log-error "Error: '$(echo $result | sed "s/.*attribute '\([^']*\)'.*/\1/")' is not available in the toolbox"
-        exit 1
-    fi
+    log "Creating shell ..."
+    commit=$(_lastCommit)
+    _generateToolboxJSON $commit
```
Comment (jeci-sarl) [inline scripts/commands.sh]: commit is not used later
```suggestion
    _generateToolboxJSON $(_lastCommit)
```

Comment (eonpatapon) [reply scripts/commands.sh]: Done

**Comment by jeci-sarl** [comment]
LGTM

**Pull request merged**